x/dsmr/node_test.go

```diff
@@ -961,11 +1000,57 @@ func TestNode_BuildBlock_IncludesChunks(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	r.NoError(node.BuildChunk(
+		context.Background(),
+		[]dsmrtest.Tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	))
+
+	blk, err := node.BuildBlock(context.Background(), node.LastAccepted, 2)
+	r.NoError(err)
+	r.NoError(node.Verify(context.Background(), node.LastAccepted, blk))
```

Comment:
I'm not sure this is testing what it's intended to. This should test that the validity window correctly eliminates chunks that are in the chunk pool from block building. However, we build, verify, and accept the block, which will remove the chunk from the chunk pool. Therefore, no chunks are actually "eliminated" by the validity window in this test.

Reply:
I added coverage for that in `TestNode_BuildBlock_IncludesChunks`

Reply:
The current flow here is build -> verify -> accept and then build another block and confirm that the chunks included in the already accepted block are correctly eliminated so `BuildBlock` returns the expected error.

At that point, the node is in a state of having an empty chunk pool and an updated last accepted block. I guess this is just testing that `BuildBlock` does not build an invalid block with the same chunk that was already included. Isn't that the same as the no available chunk certs in the test above?